src/thunderbots/software/geom/triangle.cpp

```diff
@@ -0,0 +1,87 @@
+#include "geom/triangle.h"
+
+#include <unordered_set>
+
+#include "geom/util.h"
+
+Triangle::Triangle(Point p0, Point p1, Point p2)
```

Comment:
Maybe unnecessary, but what if the points specified do not form a triangle? Like if 2 points are the same

Reply:
I suppose you'd just end up with a Triangle with 0 area. Not sure if there's much we can do to prevent it, and I don't think it's necessary.